resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/ClientBuilder.scala

```diff
@@ -51,4 +52,14 @@ private[spark] object KubernetesClientBuilder {
     }
     new DefaultKubernetesClient(clientConfigBuilder.build)
   }
+
+  def buildOkhttpClientFromWithinPod(client: BaseClient): OkHttpClient = {
+    val field = classOf[BaseClient].getDeclaredField("httpClient")
```

Comment:
why do we need this reflection to grab the httpClient out of the BaseClient?  Is this what we have to do because the fabric8 client doesn't support TPR yet?